src/libcommonserver/io/iohelper.cpp

```diff
@@ -150,6 +150,56 @@ std::string IoHelper::ioError2StdString(IoError ioError) noexcept {
     }
 }
 
+bool IoHelper::openFile(const SyncPath &path, int timeOut /*in seconds*/, std::ifstream &file, IoError &ioError) {
```

Comment:
I am hesitant about whether the default value should be 0 or 10, as 10 is required only for the special case of synced files. On the other hand, for now, this is also the only use case for this function so 🤷🏾 . To avoid any risk of regression, I explicitly provide the timeout duration for the call in `UploadJob` and in `AbstractUploadSession`.